### ISSUE TITLE
image2: throws an exception if set a content with an empty figure tag

### ISSUE THREAD

**Issue by vkmet** [ckeditor/ckeditor-dev#2506]
## Type of report

Bug

## Provide detailed reproduction steps (if any)
1.  Set extraPlugin as `config.extraPlugins = 'image2'`;
2.  Set a value for ckeditor as  `<figure class="image"></figure>`

image2 plugin throws an exception if ckeditor's content contains empty "figure" tag (or if a "figure" does not contains nor "img" nor "a" tag). 

```
<div id="editor">
    <figure class="image"></figure>
</div>

```

image2  expects that "figure" must always contains "img" tag or "a" tag that contains "img".
`image = el.getFirst( 'img' ) || el.getFirst( 'a' ).getFirst( 'img' );`
The issue can be reloved by adding additional checking:
`image = el.getFirst( 'img' ) || (el.getFirst( 'a' ) && el.getFirst( 'a' ).getFirst( 'img' ));`

### Expected result

Exception should not be thrown.

### Actual result

`Uncaught TypeError: Cannot read property 'getFirst' of null`  exception is being thrown.

## Other details

* Browser: Google Chrome 70, FireFox 62
* OS: Windows 10
* CKEditor version: ckeditor_4.10.1
* Installed CKEditor plugins: image2

**Comment by mlewand** [comment]
Actually image2 widget should not be upcasted on a `figure` that does not contain any `img` elements, as `figures` are not limited to images.

**Comment by Comandeer** [comment]
Merged into 4.11.2.
